synapse/handlers/receipts.py

```diff
@@ -206,38 +203,24 @@ def filter_out_private_receipts(
             for event_id, orig_event_content in room.get("content", {}).items():
                 event_content = orig_event_content
                 # If there are private read receipts, additional logic is necessary.
-                if (
-                    ReceiptTypes.READ_PRIVATE in event_content
-                    or ReceiptTypes.UNSTABLE_READ_PRIVATE in event_content
-                ):
```

Comment:
I think this means that we'll now leak any previous unstable private read receipts, which is a huge privacy leak.

I think we should nuke them from the database using a delta file.

Reply:
See #13692.

Reply:
Ah, sorry, I managed to forget about it again...